kratos/geometries/geometry.h

```diff
@@ -1372,21 +1372,49 @@ class Geometry
         return rResult;
     }
 
+    ///@}
+    ///@name IsInside
+    ///@{
+
     /**
-     * Returns whether given arbitrary point is inside the Geometry and the respective
-     * local point for the given global point
-     * @param rPoint The point to be checked if is inside o note in global coordinates
-     * @param rResult The local coordinates of the point
-     * @param Tolerance The  tolerance that will be considered to check if the point is inside or not
-     * @return True if the point is inside, false otherwise
-     */
+    * @brief Checks if given point in global space coordinates
+    *        is inside the geometry boundaries.
+    * @param rPointGlobalCoordinates the global coordinates of the
+    *        external point.
+    * @param rResult the local coordinates of the point.
+    * @param Tolerance the tolerance to the boundary.
+    * @return true if the point is inside, false otherwise
+    */
     virtual bool IsInside(
-        const CoordinatesArrayType& rPoint,
+        const CoordinatesArrayType& rPointGlobalCoordinates,
         CoordinatesArrayType& rResult,
         const double Tolerance = std::numeric_limits<double>::epsilon()
         ) const
     {
-        KRATOS_ERROR << "Calling base class IsInside method instead of derived class one. Please check the definition of derived class. " << *this << std::endl;
+        PointLocalCoordinates(
+            rResult,
+            rPointGlobalCoordinates);
+
+        return IsInsideLocalSpace(rResult, Tolerance);
+    }
+
+    /**
+    * @brief Checks if given point in local space coordinates of this geometry
+    *        is inside the geometry boundaries.
+    * @param rPointLocalCoordinates the point on the geometry,
+    *        which shall be checked if it lays within
+    *        the boundaries.
+    * @param Tolerance the tolerance to the boundary.
+    * @return true if the point is inside, false otherwise
+    */
+    virtual bool IsInsideLocalSpace(
```

Comment:
What the difference with IsInside?

Reply:
The local coordinates?

Reply:
IsInside requires global coordinates. For some cases this projection needs to be done outside in advance or whatever. Thus, for some scenarios using this function is more optimized.

You can check the IsInside functionhow it is defined right now, then you can see the difference.

Reply:
yep!